### PR TITLE
Add sign in and sign up with Google to Android samples

### PR DESCRIPTION
**Pull request by marinacoelho**
This PR adds a new folder to the android video samples folder. It contains the sample code for sign in and sign up with Google using Credential Manager.

Most of the new files are the same from the previous code sample (built for the anonymous authentication video), so focus on the review of: 

- `build.gradle.kts`
- `AccountService.kt`
- `AccountServiceImpl.kt`
- `AccountCenterButtons.kt`
- `SignInScreen.kt`
- `SignInViewModel.kt`
- `SignUpScreen.kt`
- `SignUpViewModel.kt`

This is how the new UI looks like:

Sign_in_with_Google
Sign_up_with_Google
Credential_Manager

### CODE REVIEW

**Review by peterfriese — COMMENTED** [review]
I'd prefer if we used the same directory structure as for Apple:

[firebase-video-samples/fundamentals/apple at main · FirebaseExtended/firebase-video-samples](https://github.com/FirebaseExtended/firebase-video-samples/tree/main/fundamentals/apple)

I assume this PR uses the new (?) Sign in with Google, so I'd be OK with "auth-sign-in-with-google".

File: fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterViewModel.kt [hunk 1]
```diff
@@ -35,8 +44,27 @@ class AccountCenterViewModel @Inject constructor(
...
+        when (val credential = result.credential) {
+            is CustomCredential -> {
+                if (credential.type == TYPE_GOOGLE_ID_TOKEN_CREDENTIAL) {
+                    linkAccountWithGoogle(credential, openScreen)
+                } else {
+                    Log.d(ERROR_TAG, UNEXPECTED_CUSTOM_CREDENTIAL)
+                }
+            }
+            else -> {
+                Log.d(ERROR_TAG, UNEXPECTED_CREDENTIAL)
```
Comment (thatfiredev) [inline fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterViewModel.kt]: Seems like these are error logs. Should we use `Log.e()` instead?

File: fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterScreen.kt [hunk 2]
```diff
@@ -107,6 +123,56 @@ fun AccountCenterScreen(
...
+            coroutineScope.launch {
+                try {
+                    val result = credentialManager.getCredential(
+                        request = request,
+                        context = context
+                    )
+
+                    viewModel.onGetCredentialResponse(result, restartApp)
+                } catch (e: GetCredentialException) {
+                    Log.d(ERROR_TAG, e.message.orEmpty())
```
[the remark this answers is not in the document]
Comment (marinacoelho) [reply fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterScreen.kt]: Yes, I'll talk about in the video, as we discussed via chat.

File: fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterViewModel.kt (hunk 1 above)
Comment (marinacoelho) [reply fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterViewModel.kt]: Good catch! Fixed.

File: fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterScreen.kt [hunk 3]
```diff
@@ -107,6 +123,56 @@ fun AccountCenterScreen(
...
+    Card(
+        modifier = Modifier.card(),
+        onClick = {
+            val googleIdOption = GetGoogleIdOption.Builder()
+                .setFilterByAuthorizedAccounts(false)
+                .setServerClientId(context.getString(R.string.default_web_client_id))
+                .build()
+
+            val request = GetCredentialRequest
+                .Builder()
```
[the remark this answers is not in the document]
Comment (marinacoelho) [reply fundamentals/android/auth-SiWG/Notes/app/src/main/java/com/notes/app/screens/account_center/AccountCenterScreen.kt]: Done.

**Review by thatfiredev — APPROVED** [review]

**Pull request merged**